test/install_script/nvm_detect_profile

```diff
@@ -1,101 +1,111 @@
 #!/bin/sh
 
-cleanup () {
-  unset -f setup cleanup die
-  unset _PROFILE
-  rm -f .bashrc .bash_profile .zshrc .profile test_profile > /dev/null 2>&1
+setup () {
+  HOME="."
+  NVM_ENV=testing . ../../install.sh
+  touch ".bashrc"
+  touch ".bash_profile"
+  touch ".zshrc"
+  touch ".profile"
+  touch "test_profile"
 }
-die () { echo $@ ; cleanup ; exit 1; }
 
-NVM_ENV=testing . ../../install.sh
-
-setup () {
-  touch .bashrc
-  touch .bash_profile
-  touch .zshrc
-  touch .profile
-  touch test_profile
+cleanup () {
+  unset HOME
+  unset NVM_ENV
+  unset NVM_DETECT_PROFILE
+  unset -f setup cleanup die
+  rm -f ".bashrc" ".bash_profile" ".zshrc" ".profile" "test_profile" > "/dev/null" 2>&1
 }
 
-#Let's hack $HOME
-HOME="."
+die () { echo "$@"; cleanup; exit 1; }
 
 setup
 
-#Let's force $SHELL to be bash
-SHELL="/bin/bash"
-
-# $SHELL is set to bash and .bashrc is there, it must be detected
-_PROFILE=$(nvm_detect_profile)
-[ "_$_PROFILE" = "_$HOME/.bashrc" ] || echo "_\$HOME/.bashrc: _$HOME/.bashrc\n" \
-                                       echo "_\$_PROFILE: _$_PROFILE\n" \
-                                       die "nvm_detect_profile didn't pick $SHELL and $HOME/.bashrc"
-
-#Let's force $SHELL to be zsh
-SHELL="/usr/bin/zsh"
-
-# $SHELL is set to zsh and .zshrc is there, it must be detected
-_PROFILE=$(nvm_detect_profile)
-[ "_$_PROFILE" = "_$HOME/.zshrc" ] || echo "_\$HOME/.zshrc: _$HOME/.zshrc\n" \
-                                      echo "_\$_PROFILE: _$_PROFILE\n" \
-                                      die "nvm_detect_profile didn't pick $SHELL and $HOME/.zshrc"
-
-
-# if we unset shell it looks for the files
-unset SHELL
-
-# $PROFILE points to a valid file, its path must be returned
-PROFILE="test_profile"
-_PROFILE=$(nvm_detect_profile)
-[ "_$_PROFILE" = "_$PROFILE" ] || echo "_\$_PROFILE: _$_PROFILE\n" \
-                                  echo "_\$PROFILE: _$PROFILE\n" \
-                                  die "nvm_detect_profile didn't pick \$PROFILE"
-
-# $PROFILE doesn't point to a valid file, its path must not be returned
-PROFILE="invalid_profile"
-_PROFILE=$(nvm_detect_profile)
-[ "_$_PROFILE" != "_$PROFILE" ] || echo "_\$_PROFILE: _$_PROFILE\n" \
-                                   echo "_\$PROFILE: _$PROFILE\n" \
-                                   die "nvm_detect_profile shouldn't pick \$PROFILE when it's not a valid file"
-
-
-# Below are tests for when $PROFILE is undefined
-rm test_profile
-unset PROFILE
+#
+# Confirm profile detection via $SHELL works and that $PROFILE overrides profile detection
+#
+
+# .bashrc should be detected for bash
+NVM_DETECT_PROFILE="$(declare SHELL="/bin/bash"; unset PROFILE; nvm_detect_profile)"
```

Comment:
It'd be ideal to simplify where possible. generally, you can set env vars for the duration of a command like `TEST="ABC" echo $TEST` without having to use separate statements. does that work here?